tutoraspects/templates/aspects/apps/aspects/migrations/alembic/versions/0024_fact_enrollments.py

```diff
@@ -0,0 +1,80 @@
+"""
+create a materialized view to populate a denormalized fact table of enrollment events
+"""
+
+from alembic import op
+
+revision = "0024"
+down_revision = "0023"
+branch_labels = None
+depends_on = None
+on_cluster = (
+    " ON CLUSTER '{{CLICKHOUSE_CLUSTER_NAME}}' "
+    if "{{CLICKHOUSE_CLUSTER_NAME}}"
+    else ""
+)
+engine = "ReplicatedMergeTree" if "{{CLICKHOUSE_CLUSTER_NAME}}" else "MergeTree"
+
+
+def upgrade():
+    op.execute(
+        f"""
+        create table if not exists {{ ASPECTS_XAPI_DATABASE }}.fact_enrollments (
+            emission_time DateTime,
+            org String,
+            course_key String,
+            course_name String,
+            course_run String,
+            actor_id String,
+            enrollment_mode LowCardinality(String),
+            enrollment_status String
```

Comment:
I guess this could be LowCardinality too since it's just the two strings?

Reply:
I think that makes sense. I don't know how many states an enrollment can be in but it seems safe to say that `LowCardinality` would give us more flexibility than an `is_enrolled` boolean while still saving space.